BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/testUtil/AbstractBeaconCreator.java

```diff
@@ -0,0 +1,56 @@
+package com.nexenio.bleindoorpositioning.testUtil;
+
+import com.nexenio.bleindoorpositioning.ble.advertising.AdvertisingPacket;
+import com.nexenio.bleindoorpositioning.ble.beacon.Beacon;
+import com.nexenio.bleindoorpositioning.ble.beacon.BeaconUtil;
+
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Use this class to create beacon with correct advertising packets in your tests.
+ *
+ * @param <B> Beacon type for which instances will be created
+ */
+public abstract class AbstractBeaconCreator<B extends Beacon> {
+
+    private Class<B> beaconClass;
+
+    public AbstractBeaconCreator(Class<B> beaconClass) {
+        this.beaconClass = beaconClass;
+    }
+
+    /**
+     * Please note that if you want a single instance of a beacon instead of adding it to a list you
+     * need to specify the complete type e.g. {@code IBeacon<IBeaconAdvertisingPacket>}.
+     *
+     * @param distance Distance for which a rssi will be generated
```

Comment:
Should be `an RSSI` I believe. Replace `generated` with `calculated`

Reply:
`An` is only used instead of `a` infront of a vowel.